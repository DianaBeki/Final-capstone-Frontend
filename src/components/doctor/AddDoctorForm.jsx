import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { createDoctor, selectStatus, selectError } from '../../redux/doctor/doctorSlice';

const AddDoctorForm = () => {
  const dispatch = useDispatch();
  const status = useSelector(selectStatus);
  const error = useSelector(selectError);

  const [formData, setFormData] = useState({
    name: '',
    specialization: '',
    image: '',
    fee: '',
    bio: '',
  });

  const handleInputChange = (e) => {
    setFormData({
      ...formData,
      [e.target.name]: e.target.value,
    });
  };

  const handleFormSubmit = (e) => {
    e.preventDefault();
    dispatch(createDoctor(formData));
  };

  return (
    <div className="max-w-md mx-auto mt-8 p-6 bg-white shadow-md rounded-md">
       <h2 className="text-2xl font-bold mb-4 text-center">Add New Doctor</h2>
      {status === 'failed' && <div className="text-red-500 mb-4 text-center">{error}</div>}
      <form onSubmit={handleFormSubmit}>
      <div className="mb-4">
          <input
            type="text"
            name="name"
            placeholder="Name"
            value={formData.name}
            onChange={handleInputChange}
            required
            className="mt-1 p-2 border rounded-md w-full focus:outline-none focus:ring focus:border-blue-300"
          />
        </div>
        <div className="mb-4">
          <input
            type="text"
            name="specialization"
            placeholder="Specialization"
            value={formData.specialization}
            onChange={handleInputChange}
            required
            className="mt-1 p-2 border rounded-md w-full focus:outline-none focus:ring focus:border-blue-300"
          />
        </div>
      </form>
    </div>
  )
}

export default AddDoctorForm;
