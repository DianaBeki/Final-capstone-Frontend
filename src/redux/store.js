import { configureStore } from '@reduxjs/toolkit';
import doctorReducer from './doctor/doctorSlice';
import userSlice from './user/userSlice';

const store = configureStore({
  reducer: {
    doctor: doctorReducer,
    user: userSlice,
  },
});

export default store;
