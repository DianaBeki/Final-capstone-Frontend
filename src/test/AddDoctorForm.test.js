import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { Provider } from 'react-redux';
import AddDoctorForm from '../components/doctor/AddDoctorForm';
import store from '../redux/store';

const render = (component) => rtlRender(
  <Provider store={store}>
    {component}
  </Provider>,
);

describe('AddDoctorForm Component', () => {
  it('renders AddDoctorForm component', () => {
    const { container } = render(<AddDoctorForm />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
