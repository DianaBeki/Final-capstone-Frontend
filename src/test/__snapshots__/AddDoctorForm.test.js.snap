// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddDoctorForm Component renders AddDoctorForm component 1`] = `
<div
  class="max-w-md mx-auto mt-8 p-6 bg-white shadow-md rounded-md"
>
  <h2
    class="text-2xl font-bold mb-4 text-center"
  >
    Add New Doctor
  </h2>
  <div
    class="mb-4"
  >
    <form>
      <div
        class="mb-4"
      >
        <input
          class="mt-1 p-2 border rounded-md w-full focus:outline-none focus:ring focus:border-blue-300"
          name="name"
          placeholder="Full name"
          required=""
          type="text"
          value=""
        />
      </div>
      <div
        class="mb-4"
      >
        <input
          class="mt-1 p-2 border rounded-md w-full focus:outline-none focus:ring focus:border-blue-300"
          name="specialization"
          placeholder="Specialization"
          required=""
          type="text"
          value=""
        />
      </div>
      <div
        class="mb-4"
      >
        <input
          class="mt-1 p-2 border rounded-md w-full focus:outline-none focus:ring focus:border-blue-300"
          name="image"
          placeholder="Image URL"
          type="text"
          value=""
        />
      </div>
      <div
        class="mb-4"
      >
        <input
          class="mt-1 p-2 border rounded-md w-full focus:outline-none focus:ring focus:border-blue-300"
          name="fee"
          placeholder="Fee"
          required=""
          type="text"
          value=""
        />
      </div>
      <div
        class="mb-4"
      >
        <textarea
          class="mt-1 p-2 border rounded-md w-full focus:outline-none focus:ring focus:border-blue-300"
          name="bio"
          placeholder="Bio"
        />
      </div>
      <button
        class="bg-green-700 hover:bg-green-400 text-white  px-4 py-2 rounded-md focus:outline-none focus:ring focus:border-green-300 w-full"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
